DEPL-11980 Investigate "OSError: [Errno 9] Bad file descriptor" when …

ashishjuyal opened the pull request:
…running XL Deploy in Docker

Jython is having issue while uploading file with the alpine linux base image. The fix is to use other distribution as base image. We tested it with jre-slim, it uses Debian as base image. The caveat is, it increases the size of the image by 100 MB and as the base image of this docker image is changed the impact will also be on the xl-se-docker-demo repository as it uses this image as base image. We need to fix those as well.

Commit:
DEPL-11980 Investigate "OSError: [Errno 9] Bad file descriptor" when running XL Deploy in Docker

Jython is having issue while uploading file with the alpine linux base image. The fix is to use other distribution as base images. We tested it with jre-slim, it uses Debian as base image. The caveat is, it increases the size of the image by 100 MB and as the base image of this docker image is changed the impact will also be on the xl-se-docker-demo repository as it uses this image as base image. We need to fix those as well.

jdewinne commented:
I would not override the current alpine Dockerfile. Instead it would be better to add 2 options: alpine and slim as docker tags.
Did you also test this with the python alpine image? Did that result in the same error?

ashishjuyal commented:
No we had not tested with python alpine. Jython does not require python and building python-alpine image with jre will result in the same issue because the issue is with Jython, jre and alpine combination.

Commit:
restoring Dockerfile

Commit:
DEPL-11980 Investigate "OSError: [Errno 9] Bad file descriptor" when running XL Deploy in Docker

added Dockerfile for slim version and updated the Readme.md for the new tag information

Commit:
DEPL-11980 Investigate "OSError: [Errno 9] Bad file descriptor" when running XL Deploy in Docker

updated the Readme

jdewinne committed:
Update README.md
